node.gyp

```diff
@@ -746,6 +746,45 @@
   ], # end targets
 
   'conditions': [
+    [ 'node_target_type=="static_library"', {
+      'targets': [
+        {
+          'target_name': 'static_node',
+          'type': 'executable',
+          'product_name': '<(node_core_target_name)',
+          'dependencies': [
+            '<(node_core_target_name)',
+          ],
+          'sources+': [
+            'src/node_main.cc',
+          ],
+          'include_dirs': [
+            'deps/v8/include',
+          ],
+          'xcode_settings': {
+            'OTHER_LDFLAGS': [
+              '-Wl,-force_load,<(PRODUCT_DIR)/libnode.a',
```

Comment:
I tried some experiments in node.gyp recently. I found if one target depends on a `static_library` target, it would add the `-force_load` to linker automatically, same as `--whole-archive` option in Linux GCC. but I am not sure about Windows. Can you double check one this? if that's the case, then no need to add these condition here.

Reply:
@bnoordhuis you are right. I found that my experimental change are mangled with Daniel's change here. And his condition is for the whole node.gyp scope. my changes are in the target scope. Although I didn't add `--whole-archive` in my target, the global scope condition still apply to my target. So just ignore my comment above. Sorry for the confusion.